sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/aio/_models.py

```diff
@@ -0,0 +1,101 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+# pylint: disable=too-few-public-methods, too-many-instance-attributes
+# pylint: disable=super-init-not-called, too-many-lines
+from azure.core.async_paging import AsyncPageIterator
+from azure.storage.blob.aio._models import ContainerPropertiesPaged
+
+from .._deserialize import return_headers_and_deserialized_path_list, process_storage_error
+from .._generated.models import StorageErrorException
+from .._models import PathPropertiesPaged as PathPropertiesPagedBase
+
+from .._models import FileSystemProperties
+
+
+class FileSystemPropertiesPaged(ContainerPropertiesPaged):
+    """An Iterable of File System properties.
+
+    :ivar str service_endpoint: The service URL.
+    :ivar str prefix: A file system name prefix being used to filter the list.
+    :ivar str marker: The continuation token of the current page of results.
+    :ivar int results_per_page: The maximum number of results retrieved per API call.
+    :ivar str continuation_token: The continuation token to retrieve the next page of results.
+    :ivar str location_mode: The location mode being used to list results. The available
+        options include "primary" and "secondary".
+    :ivar current_page: The current page of listed results.
+    :vartype current_page: list(~azure.storage.filedatalake.FileSystemProperties)
+
+    :param callable command: Function to retrieve the next page of items.
+    :param str prefix: Filters the results to return only file systems whose names
+        begin with the specified prefix.
+    :param int results_per_page: The maximum number of file system names to retrieve per
+        call.
+    :param str continuation_token: An opaque continuation token.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(FileSystemPropertiesPaged, self).__init__(
+            *args,
+            **kwargs
+        )
+
+    @staticmethod
+    def _build_item(item):
+        return FileSystemProperties._from_generated(item)  # pylint: disable=protected-access
+
+
+class PathPropertiesPaged(AsyncPageIterator, PathPropertiesPagedBase):
```

Comment:
I would remove PathPropertiesPagedBase here and just inherit from AsyncPageIterator.
It means we will have to duplicate the static method `_build_item`, but I think that's preferable to the multiple inheritance of an iterator.